components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -217,4 +227,74 @@ private void unsetCurrentSessionIdThreadLocal() {
             IdentityUtil.threadLocalProperties.get().remove(Constants.CURRENT_SESSION_IDENTIFIER);
         }
     }
+
+    private boolean validateTenantDomain(Request request, Response response, String tenantDomain)
+            throws IOException, ServletException {
+
+        try {
+            TenantManager tenantManager = AuthenticationValveServiceHolder.getInstance().getRealmService()
+                    .getTenantManager();
+            if (tenantDomain != null && !tenantManager.isTenantActive(IdentityTenantUtil.getTenantId(tenantDomain))) {
+                handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND,
+                        tenantDomain);
+                return false;
+            }
+        } catch (UserStoreException ex) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while validating tenant domain.", ex);
+            }
+            handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND, tenantDomain);
+            return false;
+        } catch (IdentityRuntimeException e) {
```

Comment:
In which cases can an IdentityRuntimeException be thrown?

Reply:
There are two cases: 

1. https://github.com/wso2/carbon-identity-framework/blob/4ed4a4366d843bf2af432e6538c3cd91031c7d32/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L265
2. https://github.com/wso2/carbon-identity-framework/blob/4ed4a4366d843bf2af432e6538c3cd91031c7d32/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L269